pkg/controller/history/controller_history.go

```diff
@@ -56,20 +56,17 @@ func ControllerRevisionName(prefix string, hash uint32) string {
 }
 
 // NewControllerRevision returns a ControllerRevision with a ControllerRef pointing to parent and indicating that
-// parent is of parentKind. The ControllerRevision has labels matching selector, contains Data equal to data, and
+// parent is of parentKind. The ControllerRevision has labels matching pod, contains Data equal to data, and
 // has a Revision equal to revision. The collisionCount is used when creating the name of the ControllerRevision
 // so the name is likely unique. If the returned error is nil, the returned ControllerRevision is valid. If the
 // returned error is not nil, the returned ControllerRevision is invalid for use.
 func NewControllerRevision(parent metav1.Object,
 	parentKind schema.GroupVersionKind,
-	selector labels.Selector,
+	podLabels map[string]string,
 	data runtime.RawExtension,
 	revision int64,
 	collisionCount *int32) (*apps.ControllerRevision, error) {
-	labelMap, err := labels.ConvertSelectorToLabelsMap(selector.String())
-	if err != nil {
-		return nil, err
-	}
+	labelMap := podLabels
```

Comment:
`podLabels` is not being used anywhere later, I agree with what you struck out. Am I missing something?

Reply:
@janetkuo pointed out in the other thread that we mutate this map below on this line:

```go
cr.Labels[ControllerRevisionHashLabel] = strconv.FormatInt(int64(hash), 10)
```

Since map types are references, this means we're mutating the map that the caller passed in to us. So it turns out we do need to make our own `labelMap` var after all, to hold a deep copy of `podLabels`. Unless you can find a utility helper for this, you'll need to just allocate a new map and then range loop over the input map to copy each entry.

Reply:
Okay, I was thinking that even if it gets mutated it's not affecting anything. But yeah, we should not mutate anything that caller passes.

Reply:
> it's not affecting anything

It actually mutates template labels (`set.Spec.Template.Labels`) the caller passes.

Reply:
Got it. Thanks.